internal/provider/openstack/provider_test.go

```diff
@@ -1189,3 +1190,60 @@ func envWithNetworking(net Networking, netCfg string) *Environ {
 		networking: net,
 	}
 }
+
+func (s *providerUnitTests) TestTerminateInstanceNetworkPorts(c *gc.C) {
```

Comment:
This isn't how we want to test this part. The go-goose upstream has test service which the nova and neutron clients can use directly without needing to stub any api calls.
The test would be added to `local_test.go`. 
I'll do the changes and push them up.